sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py

```diff
@@ -0,0 +1,82 @@
+#
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from typing import TYPE_CHECKING, Sequence
+
+from azure.core import PipelineClient
+from msrest import Deserializer, Serializer
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from typing import Any
+
+from azure.core.pipeline.policies import AzureKeyCredentialPolicy
+from azure.core.credentials import AzureKeyCredential
+from .shared_access_signature_credential import EventGridSharedAccessSignatureCredential
+from ._signature_credential_policy import EventGridSharedAccessSignatureCredentialPolicy
+from ._models import CloudEvent, EventGridEvent
+from ._generated._event_grid_publisher_client import EventGridPublisherClient as EventGridPublisherClientImpl
+from .shared_access_signature_credential import EventGridSharedAccessSignatureCredential
+from . import _constants as constants
+
+class EventGridPublisherClient(object):
+    """EventGrid Python Publisher Client.
+
+    :param str topic_hostname: The topic endpoint to send the events to.
+    :param credential: The credential object used for authentication which implements SAS key authentication or SAS token authentication.
+    :type credential: Union[~azure.core.credentials.AzureKeyCredential, azure.eventgrid.EventGridSharedAccessSignatureCredential]
+    """
+
+    def __init__(
+        self,
+        topic_hostname,  # type: str
+        credential,  # type: Union[azure.core.credential.AzureKeyCredential, azure.eventgrid.EventGridSharedAccessSignatureCredential]
+        **kwargs  # type: Any
+    ):
+        # type: (str, Union[AzureKeyCredential, EventGridSharedAccessSignatureCredential], Any) -> None
+
+        if topic_hostname[:7] == 'http://':
+            raise ValueError("Topic_hostname must either begin without the http:// or with an https://")
```

Comment:
I would slightly change tthe text saying we support only https - although we enforce it when there's no http - need to make it clear to the user